File: Java/String/HashMap/Longest Balanced Substring I.java
class Solution {
    public int longestBalanced(String s) {
        int n = s.length();
        int lbs = 0;
        for (int i = 0; i < n; i++) {
            HashMap<Character, Integer> mp = new HashMap<>();
            for (int j = i; j < n; j++) {
                mp.put(s.charAt(j), mp.getOrDefault(s.charAt(j), 0) + 1);
                int cnt = mp.get(s.charAt(j));
                boolean flag = true;
                for (Character m : mp.keySet()) {
                    if (cnt != mp.get(m)) {
                        flag = false;
                        break;
                    }
                }
                if (flag) {
                    lbs = Math.max(lbs, j - i + 1);
                }
            }
        }
        return lbs;
    }
}

File: CPP/String/Map/Longest Balanced Substring I.cpp
class Solution {
public:
    int longestBalanced(string s) {
        int n = s.size();
        int lbs = 0;
        for (int i = 0; i < n; i++) {
            unordered_map<char, int> mp;
            for (int j = i; j < n; j++) {
                mp[s[j]]++;
                int cnt = mp[s[i]];
                bool flag = true;
                for (auto m : mp) {
                    if (cnt != m.second) {
                        flag = false;
                        break;
                    }
                }
                if (flag) {
                    lbs = max(lbs, j - i + 1);
                }
            }
        }
        return lbs;
    }
};
